libraries/src/Installer/InstallerScriptTrait.php

```diff
@@ -0,0 +1,362 @@
+    }
+
+    /**
+     * Remove the files and folders in the given array from
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    protected function removeFiles(): void
+    {
+        if (!empty($this->deleteFiles)) {
+            foreach ($this->deleteFiles as $file) {
+                if (is_file(Path::check(JPATH_ROOT . $file)) && !File::delete(JPATH_ROOT . $file)) {
+                    Log::add(Text::sprintf('JLIB_INSTALLER_ERROR_FILE_FOLDER', $file));
+                }
+            }
+        }
+
+        if (!empty($this->deleteFolders)) {
+            foreach ($this->deleteFolders as $folder) {
+                if (is_dir(Path::check(JPATH_ROOT . $folder)) && !Folder::delete(JPATH_ROOT . $folder)) {
+                    Log::add(Text::sprintf('JLIB_INSTALLER_ERROR_FILE_FOLDER', $folder));
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns the internal application or null when not set.
+     *
+     * @return  CMSApplicationInterface|null
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    protected function getApplication(): ?CMSApplicationInterface
+    {
+        return $this->application;
+    }
+
+    /**
+     * Sets the application to use.
+     *
+     * @param   CMSApplicationInterface  $application  The application
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function setApplication(CMSApplicationInterface $application): void
+    {
+        $this->application = $application;
+    }
+
+    /**
+     * Creates the dashboard menu module
+     *
+     * @param string $dashboard The name of the dashboard
+     * @param string $preset    The name of the menu preset
+     *
+     * @return  void
+     *
+     * @throws \Exception
+     * @since   __DEPLOY_VERSION__
+     */
+    protected function addDashboardMenuModule(string $dashboard, string $preset)
+    {
+        $model  = $this->getApplication()->bootComponent('com_modules')->getMVCFactory()->createModel('Module', 'Administrator', ['ignore_request' => true]);
+        $module = [
+            'id'         => 0,
+            'asset_id'   => 0,
+            'language'   => '*',
+            'note'       => '',
+            'published'  => 1,
+            'assignment' => 0,
+            'client_id'  => 1,
+            'showtitle'  => 0,
+            'content'    => '',
+            'module'     => 'mod_submenu',
+            'position'   => 'cpanel-' . $dashboard,
+        ];
+
+        // Try to get a translated module title, otherwise fall back to a fixed string.
+        $titleKey         = strtoupper('COM_' . $this->extension . '_DASHBOARD_' . $dashboard . '_TITLE');
+        $title            = Text::_($titleKey);
+        $module['title']  = ($title === $titleKey) ? ucfirst($dashboard) . ' Dashboard' : $title;
+
+        $module['access'] = (int) Factory::getApplication()->get('access', 1);
```

Comment:
```suggestion
        $module['access'] = (int) $this->getApplication()->get('access', 1);
```
maybe refactory to a local $app variable?